ts/Extensions/Drilldown.ts

```diff
@@ -976,6 +976,14 @@ Chart.prototype.drillUp = function (): void {
                 }
             });
 
+            // For category, axis reset the names that the new series
+            // starts from the beginning, #16135.
+            if (oldSeries.xAxis &&
+                oldSeries.xAxis.options.type === 'category'
+            ) {
+                oldSeries.xAxis.names = [];
+            }
+
```

Comment:
I am a bit concerned that hard-checking the `type` may be too strict. And also that we don't need to reset the names, we could simply empty the existing array. We do a similar thing here:

https://github.com/highcharts/highcharts/blob/7de35360524956689d564ea6b2179ae81a12084f/ts/Extensions/Drilldown.ts#L840-L843

Can you check if the we can use the same, simpler approach in the new case?